Update RapidJSONParser.h

ivan2015112 opened the pull request:
fix 
https://github.com/ClickHouse/ClickHouse/issues/49895

<!---AI changelog entry and formatting assistance: true-->

### Changelog category (leave one):
- Improvement

When JSON contains large integer-like numbers without fractional parts, converting them to Float64 may lose precision.
This can be a problem when such numbers represent IDs.
This change detects values exceeding 2^53 and treats them as strings to preserve accuracy.

- Example use: A query or command.

before 

SELECT JSONExtractRaw('{"a": 1234567891234456456556789123456789}')  as b SETTINGS allow_simdjson = 0;
┬─b───────────────────────────┐
│ {"a":1234567891234456456000000000} │
└─────────────────────────────┴
after 

SELECT JSONExtractRaw('{"a": 1234567891234456456556789123456789}')  as b SETTINGS allow_simdjson = 0;
┬─b───────────────────────────┐
│ {"a":"1234567891234456456556789123456789"} │
└─────────────────────────────┴

ivan2015112 committed:
Update RapidJSONParser.h

fix 
https://github.com/ClickHouse/ClickHouse/issues/49895

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/ClickHouse/ClickHouse?pullRequest=89187) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/ClickHouse/ClickHouse?pullRequest=89187) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/ClickHouse/ClickHouse?pullRequest=89187) it.</sub>

ivan2015112 committed:
Update RapidJSONParser.h

ivan2015112 committed:
Update RapidJSONParser.h

add Tencent rapidjson flags

https://github.com/Tencent/rapidjson/blob/f5e5d47fac0f654749c4d6267015005b74643dff/include/rapidjson/reader.h

ivan2015112 commented on src/Common/JSONParsers/RapidJSONParser.h:
```diff
@@ -35,16 +35,23 @@ struct RapidJSONParser
+                            return ElementType::STRING; 
+                        return ElementType::DOUBLE;
+                    }
+                    return ptr->IsUint64() ? ElementType::UINT64 : ElementType::INT64;
```
@Avogar  thank you for your comment

i rollback my change and i found  fix rapidjson  from Tencent and add it here

https://github.com/Tencent/rapidjson/blob/f5e5d47fac0f654749c4d6267015005b74643dff/include/rapidjson/reader.h

Avogar replied:
I don't understand what fix you refer to. The file from your link was updated 7 years ago, we have rapidjson library in our contribs and this file is alse there. You just added a enum from the library, how it's supposed to fix the issue?

Before pushing any changes, please, first build clickhouse locally and verify that the change actually does something

ivan2015112 committed:
Update RapidJSONParser.h
